diff --git a/CMakeLists.txt b/CMakeLists.txt
index e6828d8..59aa67a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -152,6 +152,8 @@ set(RE2_HEADERS
     re2/stringpiece.h
     )
 
+set(RE2_CMAKE_INSTALL_PATH "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_CMAKEDIR}")
+
 install(FILES ${RE2_HEADERS}
         DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/re2)
 install(TARGETS re2 EXPORT re2Targets
@@ -159,12 +161,19 @@ install(TARGETS re2 EXPORT re2Targets
         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
         RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
         INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+		
+if(WIN32)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/Debug/re2.pdb
+		DESTINATION ${CMAKE_INSTALL_LIBDIR}
+		CONFIGURATIONS Debug)
+endif()
+
 install(EXPORT re2Targets
-        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/re2 NAMESPACE re2::)
+        DESTINATION ${RE2_CMAKE_INSTALL_PATH} NAMESPACE re2::)
 
 configure_package_config_file(${CMAKE_CURRENT_SOURCE_DIR}/re2Config.cmake.in
                               ${CMAKE_CURRENT_BINARY_DIR}/re2Config.cmake
-                              INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/re2
+                              INSTALL_DESTINATION ${RE2_CMAKE_INSTALL_PATH}
                               )
 write_basic_package_version_file(${CMAKE_CURRENT_BINARY_DIR}/re2ConfigVersion.cmake
                                  VERSION ${SONAME}.0.0
@@ -173,4 +182,4 @@ write_basic_package_version_file(${CMAKE_CURRENT_BINARY_DIR}/re2ConfigVersion.cm
 
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/re2Config.cmake
               ${CMAKE_CURRENT_BINARY_DIR}/re2ConfigVersion.cmake
-        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/re2)
+        DESTINATION ${RE2_CMAKE_INSTALL_PATH})
